### PR TITLE
Fix up incorrect cloudstack metadata

### PR DESCRIPTION
**Pull request by akarpik**
Fix up incorrect cloudstack metadata

### CODE REVIEW

**Comment by lamont-granquist** [comment]
my only concern would be if this changes an API and might be considered a breaking change.

@chef/client-engineers @chef/client-core needs review

**Comment by mcquin** [comment]
:+1:

**Comment by lamont-granquist** [comment]
:+1:

looks like it needs a rebase.

File: lib/ohai/plugins/cloudstack.rb [hunk 1]
```diff
@@ -31,6 +33,11 @@
         cloudstack Mash.new
         Ohai::Log.debug("connecting to the 'cloudstack' metadata service")
         fetch_metadata.each { |k, v| cloudstack[k] = v }
+        # Note: the cloudstack public_ipv4 is the value of the NAT router (as per cloudstack documentation)
+        # and not necessarily the publicly available IP.  cloustack semi-supports floating
+        # ips in that the public ip for an instance can be an IP different from the NAT router
+        cloudstack['router_ipv4'] = cloudstack.delete('public_ipv4')
```
Comment (tyler-ball) [inline lib/ohai/plugins/cloudstack.rb]: It seems like this is a compatiability-breaking change, unless it was considered a bug before and people were using workarounds

Comment (lamont-granquist) [inline lib/ohai/plugins/cloudstack.rb]: From the description it sounds like this was a bug and we were violating the documented API for cloudstack.

**Comment by btm** [comment]
rebased and merged to master.

**Pull request closed without merging**